src/sync/stitch_service_client.hpp

```diff
@@ -0,0 +1,99 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or utilied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+
+#ifndef CORE_STITCH_SERVICE_CLIENT_HPP
+#define CORE_STITCH_SERVICE_CLIENT_HPP
+
+#include <string>
+#include <realm/util/optional.hpp>
+#include <sync/generic_network_transport.hpp>
+
+namespace realm {
+namespace app {
+
+class AuthRequestClient {
+    /// Performs an authenticated request to the Stitch server, using the current authentication state, and should
+    /// throw when not currently authenticated.
+    /// @param auth_request The request to perform
+    /// @returns Response of the request
+    Response do_authenticated_request(const Request auth_request);
+};
+
+/// A class representing the authentication API routes on the Stitch server.
+class ServiceRoutes {
```

Comment:
[major] Remove all the routing functionality. It is arguably an unnecessary abstraction.

Reply:
Class removed